app/invocation/cache_requests_card.tsx

```diff
@@ -368,7 +371,62 @@ export default class CacheRequestsCardComponent extends React.Component<CacheReq
     ));
   }
 
+  private getCacheMetadata(scorecardResult: cache.ScoreCard.IResult) {
+    const digest = scorecardResult.digest;
+    const remoteInstanceName = this.props.model.getRemoteInstanceName();
+    rpc_service.service
+      .getCacheMetadata(
+        new cache.GetCacheMetadataRequest({
+          resourceName: {
+            digest: digest,
+            cacheType: scorecardResult.cacheType,
+            instanceName: remoteInstanceName,
+          },
+        })
+      )
+      .then((response) => {
+        this.setState(function (prevState, props) {
+          prevState.digestToCacheMetadata.set(digest.hash, response);
+          return {
+            digestToCacheMetadata: prevState.digestToCacheMetadata,
+          };
+        });
+      })
+      .catch((e) => {
+        console.log("Could not fetch metadata: " + BuildBuddyError.parse(e));
+
+        // Set an empty struct in the map, so FE doesn't keep trying to fetch an invalid result
+        this.setState(function (prevState, props) {
+          prevState.digestToCacheMetadata.set(digest.hash, null);
+          return {
+            digestToCacheMetadata: prevState.digestToCacheMetadata,
+          };
+        });
+      });
+  }
+
   private renderResultHovercard(result: cache.ScoreCard.IResult, startTimeMillis: number) {
+    const digest = result.digest.hash;
+    if (!this.state.digestToCacheMetadata.has(digest)) {
```

Comment:
It's usually a best practice for `render()` to not have side effects, but there's not really a good alternative atm. Mind adding this TODO?

```suggestion
    // TODO(bduffany): Add an `onHover` prop to <Tooltip> and move this logic there
    if (!this.state.digestToCacheMetadata.has(digest)) {
```

Reply:
Done